tests/login.test.js

```diff
@@ -0,0 +1,40 @@
+
+import loginPage from '../pages/login.page';
+import homePage from '../pages/home.page';
+import { expect } from 'chai';
+
+describe('OpenBank', () => {
+
+    
+    it('Should access the home page when attempting to login with valid credentials', async () => {
+        await loginPage.open('login');
+        let username = 'jsmith@demo.io';
+        let password = 'Demo123!';
+        await loginPage.enter(username, password);
+       
+        
+        expect(await homePage.getDashboardTitle()).to.equal('Dashboard');
+        
+     });
+     it('Should show an error when attempting to login with invalid credentials', async () => {
+        await loginPage.open('login');
+        
+        let username = 'asdfasdf';
+        let password = '123456';
+        await loginPage.enter(username, password);
+        
+        expect(await browser.getUrl()).to.equal('http://localhost:8080/bank/login?error');
+        
+     });
+     it('Should show an error when attempting to login with blank credentials', async () => {
+        await loginPage.open('login');
+        
+        let username = '';
+        let password = '';
+        await loginPage.enter(username, password);
+        
+        expect(await browser.getUrl()).to.equal('http://localhost:8080/bank/login?error');
```

Comment:
Prefer this implementation, as it's more readable and doesn't require to pass the absolute path (this makes tests independent of environment the application is running on, as current implementation only checks for URLs of a locally hosted application.)

```javascript
await expect(browser).toHaveUrlContaining('login?error');
```

Reply:
the same above...